### PR TITLE
Bunk beds

### PR DESCRIPTION
**Pull request by RarkGrames**
#### Summary
```SUMMARY: Content "Added a bunk bed furniture"```

#### Purpose of change
Breaking down #37024 to smaller portions for easier review.

#### Describe the solution
- Adding the furniture
- Replacing existing beds in the `bunker` and `outpost` mapgen files

#### Testing
Spawned the locations and checked out if they fit the places and spawn at all.

### CODE REVIEW

**Comment by curstwist** [comment]
might want to add construction recipes for these new furniture pieces when appropriate,

**Comment by RarkGrames** [comment]
Added recipe.

File: data/json/construction.json [hunk 1]
```diff
@@ -1703,6 +1703,17 @@
...
     "post_terrain": "f_bed"
   },
+  {
+    "type": "construction",
+    "description": "Build Bunk Bed",
+    "category": "FURN",
+    "required_skills": [ [ "fabrication", 4 ] ],
+    "time": "130 m",
+    "qualities": [ [ { "id": "HAMMER", "level": 2 } ], [ { "id": "SAW_W", "level": 1 } ] ],
+    "components": [ [ [ "2x4", 30 ] ], [ [ "nail", 30 ] ], [ [ "mattress", 2 ], [ "down_mattress", 2 ] ] ],
```
Comment (I-am-Erk) [inline data/json/construction.json]:
```suggestion
    "components": [ [ [ "2x4", 16 ], [ "wood_panel", 4 ] ], [ [ "2x4", 14 ] ], [ [ "nail", 30 ] ], [ [ "mattress", 2 ], [ "down_mattress", 2 ] ] ],
```
This lets them choose to use 2x4 or wood panels for the slats of the bed. Variety is the spice of life!

File: data/json/furniture_and_terrain/furniture-sleep.json [hunk 2]
```diff
@@ -27,6 +27,37 @@
...
+    "color": "magenta",
+    "move_cost_mod": 3,
+    "coverage": 40,
+    "comfort": 5,
+    "floor_bedding_warmth": 1000,
+    "required_str": -1,
+    "deconstruct": {
+      "items": [ { "item": "mattress", "count": 2 }, { "item": "2x4", "count": 28 }, { "item": "nail", "charges": [ 20, 30 ] } ]
+    },
+    "flags": [ "TRANSPARENT", "FLAMMABLE_ASH", "ORGANIC", "MOUNTABLE", "CAN_SIT", "EASY_DECONSTRUCT" ],
```
[the remark this answers is not in the document]
Comment (RarkGrames) [reply data/json/furniture_and_terrain/furniture-sleep.json]: I will add a "frame" and a "place mattress" recipe.

File: data/json/mapgen/bunker.json [hunk 3]
```diff
@@ -267,7 +267,7 @@
         "3": "f_speaker_cabinet"
       },
       "items": {
-        "b": { "item": "bed", "chance": 60 },
+        "b": { "item": "bunkbed", "chance": 60 },
```
Comment (BevapDin) [inline data/json/mapgen/bunker.json]: Where is that item group actually defined?

**Comment by RarkGrames** [comment]
@BevapDin I PR'd a fix for it.

**Pull request merged**